lean4-infoview/src/infoview/infos.tsx

```diff
@@ -69,38 +88,38 @@ export function Infos() {
 
     // Update pins on UI actions
     const pinKey = React.useRef<number>(0);
-    const isPinned = (pinnedPoss: DocumentPosition[], pos: DocumentPosition) => {
-        return pinnedPoss.some(p => DocumentPosition.isEqual(p, pos));
+    const isPinned = (pinnedPositions: DocumentPosition[], pos: DocumentPosition) => {
+        return pinnedPositions.some(p => DocumentPosition.isEqual(p, pos));
     }
     const pin = React.useCallback((pos: DocumentPosition) => {
-        setPinnedPoss(pinnedPoss => {
-            if (isPinned(pinnedPoss, pos)) return pinnedPoss;
+        setPinnedPositions(pinnedPositions => {
+            if (isPinned(pinnedPositions, pos)) return pinnedPositions;
             pinKey.current += 1;
-            return [ ...pinnedPoss, { ...pos, key: pinKey.current.toString() } ];
+            return [ ...pinnedPositions, { ...pos, key: pinKey.current.toString() } ];
         });
     }, []);
     const unpin = React.useCallback((pos: DocumentPosition) => {
-        setPinnedPoss(pinnedPoss => {
-            if (!isPinned(pinnedPoss, pos)) return pinnedPoss;
-            return pinnedPoss.filter(p => !DocumentPosition.isEqual(p, pos));
+        setPinnedPositions(pinnedPositions => {
+            if (!isPinned(pinnedPositions, pos)) return pinnedPositions;
+            return pinnedPositions.filter(p => !DocumentPosition.isEqual(p, pos));
         });
     }, []);
 
     // Toggle pin at current position when the editor requests it
     useEvent(ec.events.requestedAction, act => {
         if (act.kind !== 'togglePin') return
         if (!curPos) return
-        setPinnedPoss(pinnedPoss => {
-            if (isPinned(pinnedPoss, curPos)) {
-                return pinnedPoss.filter(p => !DocumentPosition.isEqual(p, curPos));
+        setPinnedPositions(pinnedPositions => {
+            if (isPinned(pinnedPositions, curPos)) {
+                return pinnedPositions.filter(p => !DocumentPosition.isEqual(p, curPos));
             } else {
                 pinKey.current += 1;
-                return [ ...pinnedPoss, { ...curPos, key: pinKey.current.toString() } ];
```

Comment:
Can you explain to me what the key is for? I see each pin is getting a unique key, but why? I don't really see the key being used anywhere.  Everything is using DocumentPosition.isEqual(p, pos) to compare saved pinned positions, nothing is comparing the key.  Or is it just to trigger a render in React when the pinKey.current value is incremented?

Reply:
Definitely! The `key` property is required by React to uniquely identify components within a list. It is not used by the components themselves. It is explained [here](https://reactjs.org/docs/lists-and-keys.html#keys).